src/card_comparator.ts

```diff
@@ -0,0 +1,25 @@
+export class CardComparator {
+  compare(card1: Card, card2: Card): int {
+    return undefined;
+  }
+}
+
+export class TrumpAndLeadCardComparator extends CardComparator {
+  constructor(trump: Suit, lead: Suit, rankComparator: RankComparator) {
```

Comment:
I really don't like this.  At the very least, the lead suit being part of the constructor a) is odd, and b) means that you'll need to construct a new one at least once per trick.

I don't think the comparator really cares about trump or lead suit; I think the comparator cares about a rank comparator and a suit comparator.